src/components/shared/FormInfo.js

```diff
@@ -32,54 +40,103 @@ const FormInfo = (props) => {
     nameButton,
   } = props;
 
-  const initialState = useMemo(
-    () => ({
-      title: '',
-      release_date: '',
-      poster_path: '',
-      genres: '',
-      overview: '',
-      runtime: '',
-    }),
-    []
-  );
+  const initialValues = {
+    id: '',
+    title: '',
+    date: '',
+    url: '',
+    genre: '',
+    overview: '',
+    runtime: '',
+  };
 
-  const [selectedOption, setSelectedOption] = useState(null);
+  const FormSchema = Yup.object({
+    id: Yup.number().integer('Invalid format'),
+    title: Yup.string().required('Required'),
+    date: Yup.date().required('Required'),
+    url: Yup.string().url('Invalid format').required('Required'),
+    genre: Yup.string().nullable().required('Required'),
+    overview: Yup.string().required('Required'),
+    runtime: Yup.number()
+      .integer('Invalid format')
+      .positive('Invalid format')
+      .required('Required'),
+  });
 
-  const setSelectValue = useCallback(() => {
-    const { genres } = filmEdit;
+  const updateFilms = (films, editFilm) => {
+    let objMovies = { ...films };
 
-    if (genres.length) {
-      let selectedValue = [];
-      selectOptions.forEach((item) => {
-        if (genres.includes(item.label)) {
-          selectedValue.push(item);
+    objMovies.data.forEach((item) => {
+      if (item.id === editFilm.id) {
+        for (const key in item) {
+          if (key !== 'id') {
+            item[key] = editFilm[key];
+          }
         }
-      });
-
-      if (selectedValue.length) {
-        setSelectedOption(selectedValue);
       }
-    }
-  }, [filmEdit]);
+    });
 
-  const [dataForm, setData] = useState(initialState);
+    return objMovies;
+  };
+
+  const formik = useFormik({
+    initialValues,
+    validationSchema: FormSchema,
+
+    onSubmit: (values) => {
+      let newFilm = {
+        title: values.title,
+        genres: Array.from(values.genre, (item) => item.label),
+        release_date: values.date,
+        poster_path: values.url,
+        overview: values.overview,
+        runtime: values.runtime,
+      };
+
+      if (nameButton === 'Submit') {
+        dispatch(addMovie(newFilm));
+        handleClose();
+      } else if (nameButton === 'Save') {
+        newFilm.id = values.id;
+        newFilm.vote_average = filmEdit.vote_average;
+
+        dispatch(updateMovie(newFilm));
+        dispatch(setMoviesByGenre(updateFilms(moviesByCriteria, newFilm)));
+        dispatch(fetchMoviesSuccess(updateFilms(movies, newFilm)));
```

Comment:
I wonder why do you need three separate action for this here?
We are usually handle such cases in actions.

Reply:
fix